third_party/terraform/resources/resource_google_service_account.go

```diff
@@ -131,29 +137,44 @@ func resourceGoogleServiceAccountDelete(d *schema.ResourceData, meta interface{}
 
 func resourceGoogleServiceAccountUpdate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
+
 	sa, err := config.clientIAM.Projects.ServiceAccounts.Get(d.Id()).Do()
 	if err != nil {
 		return fmt.Errorf("Error retrieving service account %q: %s", d.Id(), err)
 	}
+
+	serviceAccount := &iam.ServiceAccount{
+		Etag: sa.Etag,
+	}
 	updateMask := make([]string, 0)
+
 	if d.HasChange("description") {
+		if v, ok := d.GetOk("description"); ok {
+			serviceAccount.Description = v.(string)
+		}
 		updateMask = append(updateMask, "description")
 	}
+
 	if d.HasChange("display_name") {
+		if v, ok := d.GetOk("display_name"); ok {
+			serviceAccount.DisplayName = v.(string)
+		}
 		updateMask = append(updateMask, "display_name")
 	}
+
+	if len(updateMask) == 0 {
+		return nil
```

Comment:
Can you also put something in the debug logs that we found nothing to update? We shouldn't be able to get here unless there was actually a change.